acceptance/api/v1/suite_test.go

```diff
@@ -63,13 +60,14 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 	// install to do it's thing without needing the patch script to run first.
 	// Eventually is used to retry in case the rollout of the patched deployment
 	// is not completely done yet.
+	epinioBinary := testenv.EpinioBinaryPath()
 	fmt.Println("Ensure default workspace exists")
 	testenv.EnsureDefaultWorkspace(epinioBinary)
```

Comment:
Having a default namespace is part of installing Epinio. If we take care of patching outside the pipeline, I think we should take care of creating the missing namespace there too.

Now that I look at it again, we are mixing things a bit. For example, we don't install Epinio and we don't patch the epinio server pod but we do compile the client binary and we try to create the missing namespace here. What if the cluster we are talking to, is running a different version of the Epinio server or missing components (because it's an "old" cluster for example)?

In my opinion, we should either do everything in the pipeline (like we did) or nothing at all (which is what this PR should be all about). Doing nothing means:

- Compile/Install/Patch Epinio outside the pipeline. Also create the missing default namespace if needed.
- Pass a kubeconfig, an epinio config and an epinio binary path to the pipeline
- The pipeline runs the tests using the above three, assuming they all match the epinio version which is tested.

With this plan, consistency is a responsibility of the developer or the pipeline and not the test suite.

Another option would be, to still have a helper do the first step in go code and call it conditionally in the before suite block, if the information of step 2 is not passed to the pipeline. I think this is what we tried to do initially but living next to the test code made it too easy to spread. After a certain point, the setup of the cluster was all over the place. That's why I believe the setup should better be handled outside the acceptance test suite.

Of course the installation tests are a different matter.

Reply:
Agreed.